evm/vm/code_stream.py

```diff
@@ -66,28 +68,35 @@ def seek(self, pc):
     _validity_cache = None
 
     def is_valid_opcode(self, position):
+        # if position longer than bytecode return false
         if position >= len(self):
             return False
 
-        if position not in self._validity_cache:
-            with self.seek(max(0, position - 32)):
-                prefix = self.read(min(position, 32))
-
-            for offset, opcode in enumerate(reversed(prefix)):
-                if opcode < opcode_values.PUSH1 or opcode > opcode_values.PUSH32:
-                    continue
-
-                push_size = 1 + opcode - opcode_values.PUSH1
-                if push_size <= offset:
-                    continue
-
-                opcode_position = position - 1 - offset
-                if not self.is_valid_opcode(opcode_position):
-                    continue
-
-                self._validity_cache[position] = False
-                break
+        # return false if position already in val_cache
+        if position in self._validity_cache:
+            return False
+        # return true if position not in val_cache but has been parsed
+        if position <= self.deepest:
+            return True
+        else:
+            # set counter to deepest parsed position
+            i = self.deepest
+            while i <= position:
+                # get opcode
+                with self.seek(i):
```

Comment:
It'd be nice to do this without seeking into it.  Can we just add an `__getitem__` to this object so you can just do `self[i]` to grab the value at the given length?